### DISCUSSION TITLE
Incorrect behaviour for MockDatabase transaction log

### DISCUSSION THREAD

**Discussion by TommyLike** [SeaQL/sea-orm#1877] [General]
## Description
When using mockdatabase within transaction, observed two possible issues:
1. It will produce incorrect database backend in begin transaction log
2. unable to create identical statement by library

Check the example below:
```
  let db = MockDatabase::new(DatabaseBackend::MySql)
            .append_exec_results([
                MockExecResult{
                    last_insert_id: 1,
                    rows_affected: 1,
                }
            ]).into_connection();

        let datakey_repository = DataKeyRepository::new(&db);
        let mut tx = db.begin().await?;
       //some operation on tx
        tx.commit().await?;
       //print the tx transaction
       Transaction { stmts: [
       1. Statement { sql: "BEGIN", values: None, db_backend: Postgres }, 
       2. other statement according to the upper operation
       4. Statement { sql: "COMMIT", values: None, db_backend: MySql }] }]
```
**question**:
1. begin statement show it's configured with posgres backend
2. unable to create identical statement cause the values in transaction is "None" while self build statement will always have something like "Some(Values([]))"

## Steps to Reproduce

1. 
5. 
6.

**Comment by tyt2y3** [comment]
I don't really understand the issue, can you submit a reproducible example?

**Comment by TommyLike** [comment]
> I don't really understand the issue, can you submit a reproducible example?

ok, first I think the begin and commit statement with my configuration should always show it's configured with `mysql` backend rather than the `postgres` sql, second I can not assert the begin and commit statement cause I failed to create the statement which will have value equal to `None`
